index.js

```diff
@@ -1,19 +1,31 @@
-'use strict'
+'use strict';
 
-const core = require('@actions/core')
-const { GitHub, context } = require('@actions/github')
+const core = require('@actions/core');
+const { GitHub, context } = require('@actions/github');
 
 const main = async () => {
-  const token = core.getInput('github-token')
-  const number = core.getInput('number')
+  const token = core.getInput('github-token');
+  const approveLabel = core.getInput('approve-label');
 
-  const octokit = new GitHub(token)
+  const pullRequest = context.payload.pull_request;
+  if (!pullRequest) {
+    console.log("Could not get pull request from context, exiting");
+    return;
+  }
+
+  const labels = pullRequest.labels.map(label => label.name);
+  if (!labels.includes(approveLabel)) {
+    console.log("PR does not include approve-label, exiting");
+    return;
+  }
```

Comment:
I think you're changing this action to something else, it currently is used for approving arbitrary PRs by number, and it looks like you're proposing to change it to one that will auto-approve PRs if they contain a certain label.

Last time I checked there are already actions for this on the marketplace.